doc/user/client.md

```diff
@@ -0,0 +1,411 @@
+# Operator-SDK: Controller Runtime Client API
+
+## Overview
+
+The [`controller-runtime`][repo-controller-runtime] library provides various abstractions to watch and reconcile resources in a k8s cluster via CRUD (Create, Update, Delete, as well as Get and List in this case) operations. Operators use at least one controller to perform a coherent set of tasks within a cluster, usually through a combination of CRUD operations. The Operator SDK uses controller-runtime's [Client][doc-client-client] interface, which defines a Reader and Writer to perform these operations.
```

Comment:
`which defines a Reader and Writer to perform these operations`
Since we don't really explain or go over the Reader and Writer interfaces, nor do we need to since they're internal interfaces, we can reword this to:
`which provides the interface for CRUD operations`